File: online-judge-solutions/Codeforces/551A.py
#https://codeforces.com/problemset/problem/551/A
#Lang		:	Python 3.8
#Time		:	436 ms
#Memory         :	200 KB
n=int(input())
a=list(map(int,input().split()))
m=p={}
for i in a:m[i]=m.get(i,0)+1
m=dict(sorted(m.items()))
m[list(m)[-1]+1]=0
for i in range(1,len(m),1):
    s=1
    for k,v in list(m.items())[i:len(m)]:s+=v
    p[list(m.keys())[i-1]]=s
for i in a:print(p[i],end=" ")

File: online-judge-solutions/Codeforces/551A.cpp
//https://codeforces.com/problemset/problem/551/A
//Lang		:	GNU C++14
//Time		:	46 ms
//Memory	:	300 KB
#include<iostream>
#include<map>
using namespace std;
int main(){
	int n,x;cin>>n;
	int a[n];
	map<int,int> m,p;
	for(int i=0;i<n;i++){cin>>a[i];m[a[i]]++;}
	m[m.rbegin()->first+1]=0;
	for(int i=1;i<m.size();i++){
		int s=1;
		for(auto j=next(m.begin(),i);j!=--m.end();j++) s+=j->second;
		p[next(m.begin(),i-1)->first]=s;
	}
	for(int i:a){cout<<p[i]<<" ";}cout<<endl;
	return 0;
}
